pkg/actions/lua/lakefs/catalogexport/hive.lua

```diff
@@ -0,0 +1,101 @@
+local regexp = require("regexp")
+local pathlib = require("path")
+local common = require("lakefs/catalogexport/common")
+
+-- extract partition prefix from full path
+function extract_partitions_path(partition_cols, path)
+    local idx = 0
+    for _, partition in ipairs(partition_cols) do
+        local pattern = partition .. "=[^/]*"
+        local re = regexp.compile(pattern)
+        local match = re.find(path, pattern)
+        if match == "" then
+            return nil
+        end
+        -- expanding the pattern to a match regex because string.find() does not implement pattern matching https://github.com/Shopify/go-lua/blob/main/string.go#L37  
+        local i, j = string.find(path, match, idx)
```

Comment:
If this is the case and we require to compile the pattern - I suggest to update the implementation and match the partition + "=" and lookup for "/" after a successful match.
It will be much better than compile expression for each extract.

Reply:
changed that to call the compile once only.